website/server/controllers/api-v3/auth.js

```diff
@@ -584,8 +608,14 @@ api.updateEmail = {
 
     if (emailAlreadyInUse) throw new NotAuthorized(res.t('cannotFulfillReq'));
 
-    let candidatePassword = passwordUtils.encrypt(req.body.password, user.auth.local.salt);
-    if (candidatePassword !== user.auth.local.hashed_password) throw new NotAuthorized(res.t('wrongPassword'));
+    let password = req.body.password;
```

Comment:
I've tried this approach but at the end decided against it because it was even less straightforward because there are maximum two places where the exact same code could be used, the others needed some tweaks